Getting started guides for Kyma (#9296)
kyma-project/kyma: docs/getting-started/03-deploy-microservice.md

---
title: Deploy a microservice
type: Getting Started
---

> You will now start by deploying a standalone [`orders-service`](https://github.com/kyma-project/examples/blob/master/http-db-service/README.md) microservice in the `orders-service` Namespace. This microservice will act as a link between the external application and the Redis service and we will build the whole end-to-end flow around it.

Reviewer: ```suggestion
You will deploy a standalone [`orders-service`](https://github.com/kyma-project/examples/blob/master/http-db-service/README.md) microservice in the `orders-service` Namespace. This microservice will act as a link between the external application and the Redis service and we will build the whole end-to-end flow around it.
```
**Reason:**
`You will now start by (...)` suggest that this is just the first step to take, while in reality this is the goal that we want to achieve with this guide. 
I believe this sentence means to introduce the reader into the purpose of this guide and therefore simply telling them what they'll do in this guide using `You will now deploy (...)` seems more suitable here.

Author: I mean it in the context of the whole set - in this sense this guide is only a start, but again (same as before), if you think that might be unclear, sure I will change that.

In this guide you will create:

- Deployment in which you specify the configuration of your microservice.

> - Kubernetes Service through which your microservice will communicate with other resources on the cluster.

Reviewer: ```suggestion
- Kubernetes service through which your microservice will communicate with other resources on the cluster.
```
I believe `service` is not capitalized in our documentation. See the documentation [here](https://kyma-project.io/docs/components/application-connector/#architecture-application-connector-components-application-registry) and [here](https://kyma-project.io/docs/components/service-mesh/#overview-overview).
If it's our mistake, we should log an issue for that.

Author: Sure, in that case we should log a follow-up issue for updating our docs as k8s Service should be capitalized as all k8s resources.

Reviewer: Done :) 
[Added](https://github.com/kyma-project/community/pull/504) it to the [guidelines](https://github.com/kyma-project/community/blob/master/guidelines/content-guidelines/08-style-and-terminology.md#capitalized-terms) and logged an [issue](https://github.com/kyma-project/kyma/issues/9368) for that.

## Steps

### Create the Deployment

> Create a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) that provides the microservice definition and enables you to run it on the cluster. The Deployment uses the `eu.gcr.io/kyma-project/pr/orders-service:PR-162` image. This Docker image exposes the `8080` port on which the related Service is listening.

Reviewer: ```suggestion
Create a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) that provides the microservice definition and enables you to run it on the cluster. The Deployment uses the `eu.gcr.io/kyma-project/pr/orders-service:PR-162` image. This Docker image exposes the `8080` port on which the related service is listening.
```

See the comment above.

Follow these steps:

<div tabs name="steps" group="deploy-microservice">
  <details>
  <summary label="kubectl">
  kubectl
  </summary>

1. Apply the microservice definition to the `orders-service` Namespace on your cluster:

```bash
kubectl apply -f https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-deployment.yaml
```

2. Check if the Deployment was created. The correct Deployment status should set **readyReplicas** to `1`:

```bash
kubectl get deployment orders-service -n orders-service -o=jsonpath="{.status.readyReplicas}"
```

</details>
<details>
<summary label="ui">

> UI

Reviewer: ```suggestion
Console UI
```

</summary>

> 1. Create `orders-service-deployment.yaml` on your machine containing [this Deployment definition](https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-deployment.yaml).

Reviewer: ```suggestion
1. On your machine, create `orders-service-deployment.yaml` containing [this Deployment definition](https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-deployment.yaml).
```
**Reason:**
I restructured that to avoid ambiguity (i.e. the deployment definition is contained by the `.yaml` file or the machine?). 
I thought about wording it like this first: `1. Create `orders-service-deployment.yaml` containing [this Deployment definition](https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-deployment.yaml) on your machine.` but this is still a bit ambiguous.

Author: Ok. I change it this way (both for Deployment and Service):
```
1. On your machine, create  ...
2. Back in the Console UI, go to ... 
```

> 2. Back in the Console UI, go to the `orders-service` Namespace overview, select the **Deploy new resource** button.

Reviewer: ```suggestion
2. Back in the Console UI, go to the `orders-service` Namespace overview and select the **Deploy new resource** button.
```

3. Browse the `orders-service-deployment.yaml` file and select **Deploy** to confirm the changes.
4. Go to **Operation** > **Deployments** to make sure the status of `orders-service` is `RUNNING`.

</details>
</div>

### Create the Service

> Deploy the Kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/) in the `orders-service` Namespace to allow other Kubernetes resources to communicate with your microservice.

Reviewer: ```suggestion
Deploy the Kubernetes [service](https://kubernetes.io/docs/concepts/services-networking/service/) in the `orders-service` Namespace to allow other Kubernetes resources to communicate with your microservice.
```
See the comment above.

Follow these steps:

<div tabs name="steps" group="deploy-microservice">
  <details>
  <summary label="kubectl">
  kubectl
  </summary>

> Apply the Kubernetes Service to the `orders-service` Namespace on your cluster:

Reviewer: ```suggestion
Apply the Kubernetes service to the `orders-service` Namespace on your cluster:
```

See the comment above.

```bash
kubectl apply -f https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-service.yaml
```

  </details>
  <details>
  <summary label="ui">

>   UI

Reviewer: ```suggestion
  Console UI
```

  </summary>

> 1. Create `orders-service-service.yaml` on your machine containing [this Service definition](https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-service.yaml).

Reviewer: ```suggestion
1. On your machine, create `orders-service-service.yaml` containing [this Service definition](https://raw.githubusercontent.com/kyma-project/examples/master/orders-service/deployment/orders-service-service.yaml).
```

See the comment above.

> 2. Back in the Console UI, go to the `orders-service` Namespace overview, select the **Deploy new resource** button.

Reviewer: ```suggestion
2. Back in the Console UI, go to the `orders-service` Namespace overview and select the **Deploy new resource** button.
```
